### PR TITLE
Replace LYS coming soon override to use template include filter

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

- `parse_request` is heavy-handed is not the best practice to override templates. It also forces us to terminate the request to avoid other templates to render which may not be desirable. This PR updates the method to override coming soon template to use `template_include` filter instead.
- Now any non-existant paths would revert to WordPress' default handling which is 404.

### How to test the changes in this Pull Request:

1. Go to WooCommerce > Settings > Site Visibility
2. Set the site to `Coming soon` and save
3. Preview your front end and observe that it shows `Pardon our dust! We're working on something amazing — check back soon!`
8. Navigate to the path `/dkfisodfjwerw` from your site
9. Observe that it shows `Page Not Found`
4. Go back to Site Visibility settings and enable "Restrict to store pages only"
5. Preview any store page such as `/shop` and observe it shows `Great things coming soon`
6. Navigate to the path `/ch` from your site
7. Observe that it redirects to `/cart`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/WCAdminHelper.php [hunk 1]
```diff
@@ -238,7 +238,7 @@ private static function get_normalized_url_path( $url ) {
 		$query           = wp_parse_url( $url, PHP_URL_QUERY );
 		$path            = wp_parse_url( $url, PHP_URL_PATH ) . ( $query ? '?' . $query : '' );
 		$home_path       = wp_parse_url( site_url(), PHP_URL_PATH );
-		$normalized_path = trim( substr( $path, strlen( $home_path ) ), '/' );
+		$normalized_path = trim( substr( $path, strlen( $home_path ?? '' ) ), '/' );
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Admin/WCAdminHelper.php:241]: This was giving an error `Deprecated: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/wp-content/plugins/woocommerce/src/Admin/WCAdminHelper.php on line 241`

**Comment by adrianduffell** [comment]
Nice improvements, @ilyasfoo! By removing the `die`, I think we could now unskip the [tests](https://github.com/woocommerce/woocommerce/blob/9a288fb4c25dc46591102ac227101f7f969a6f1e/plugins/woocommerce/tests/php/src/Internal/ComingSoon/ComingSoonRequestHandlerTest.php) (with a few tweaks to account for the hook change).

**Comment by ilyasfoo** [comment]
@adrianduffell Good call, I'll re-enable the tests!

**Comment by ilyasfoo** [comment]
It seems like this solution did not work for classic themes, closing this PR until we figure out a different way to handle tests. Likely will resolve in a different PR altogether

**Pull request closed without merging**